modules/dag-l1/src/main/scala/org/tessellation/dag/l1/domain/block/BlockRelations.scala

```diff
@@ -0,0 +1,32 @@
+package org.tessellation.dag.l1.domain.block
+
+import cats.effect.kernel.Async
+import cats.syntax.eq._
+import cats.syntax.functor._
+import cats.syntax.traverse._
+
+import org.tessellation.dag.domain.block.DAGBlock
+import org.tessellation.kryo.KryoSerializer
+import org.tessellation.schema.{BlockReference, transaction}
+import org.tessellation.security.Hashed
+import org.tessellation.security.signature.Signed
+
+object BlockRelations {
+  def areRelated[F[_]: Async: KryoSerializer](block: Hashed[DAGBlock]): F[Signed[DAGBlock] => Boolean] =
+    areRelated(Set(block), Set.empty)
+
+  def areRelated[F[_]: Async: KryoSerializer](
```

Comment:
what about sth like this?

```scala
  def isChildrenOf[F[_]: Async: KryoSerializer](
    parents: Set[Hashed[DAGBlock]],
    parentsReferences: Set[BlockReference]
  )(block: Signed[DAGBlock]): F[Boolean] =
     parents.toList
        .flatTraverse(_.transactions.toSortedSet.toList.traverse(TransactionReference.of(_)))
        .map(_.toSet)
        .flatMap { txRefs =>
          def dstAddresses = parents.flatMap(_.transactions.toSortedSet.toList.map(_.value.destination))

          def isChild = block.parent.exists(parentRef => (parents.map(_.ownReference) ++ parentsReferences).exists(_ === parentRef))
          def hasReferencedTx =  block.transactions.map(_.parent).exists(txnParentRef => txRefs.exists(_ === txnParentRef))
          def hasReferencedAddress = block.transactions.map(_.source).exists(srcAddress => dstAddresses.exists(_ === srcAddress))
        
          isChild || hasReferencedTx || hasReferencedAddress
        }
```

Reply:
then you can use it like:

isChildrenOfHashedBlocks = BlockRelations. isChildrenOf(hashedBlocks)

Reply:
One block is not necessarily a child of another, but there may still be influence between them in some cases.

Reply:
Solution above doesn't work because of implicits parameters. It looks like closure cannot be done on them.

Reply:
@kriscieslak anyway, the rest of the function can be changed to clearly see the conditions for example